lib/internal/Magento/Framework/View/Element/AbstractBlock.php

```diff
@@ -665,6 +665,15 @@ public function toHtml()
         }
         $html = $this->_afterToHtml($html);
 
+        /** @var \Magento\Framework\DataObject */
+        $transportObject = new \Magento\Framework\DataObject(
```

Comment:
Thank you for your comment.

I followed the example of `view_block_abstract_to_html_before` event which is dispatched at the beginning of the `toHtml()` method: in what does the approach of dispatching this event differ from mine?

Regarding your suggestion of putting the event in the `\Magento\Framework\View\Layout::renderNonCachedElement` do you suggest to add both a `before` and an `after` event? 

Furthermore, would you suggest to differentiate the event based on the element (UI component, Block or Container) or should I keep the same event for all three?

Cheers,
Alessandro